apps/frontend/src/app/2_molecules/SystemStats/hooks/useGetAssetBalance.ts

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react';
+
+import { Contract } from 'ethers/lib/ethers';
+
+import { getTokenDetails, getTokenContract } from '@sovryn/contracts';
+import { SupportedTokens } from '@sovryn/contracts';
+import { getProvider } from '@sovryn/ethers-provider';
+
+import {
+  idHash,
+  asyncCall,
+  CacheCallResponse,
+} from '../../../../store/rxjs/provider-cache';
+import { getRskChainId } from '../../../../utils/chain';
+import { EcosystemDataType } from '../types';
+
+export const useGetAssetBalance = (
+  asset: SupportedTokens,
+  contractToken: string,
+  chainId = getRskChainId(),
+): CacheCallResponse<string> => {
+  const [state, setState] = useState({
+    value: '0',
+    loading: false,
+    error: null,
+  });
+
+  useEffect(() => {
+    const getBalance = async () => {
+      try {
+        const { address: tokenAddress } = await getTokenContract(
+          contractToken,
+          chainId,
+        );
+        const tokenDetails = await getTokenDetails(asset, chainId);
+
+        const hashedArgs = idHash([
+          tokenDetails.address,
+          EcosystemDataType.balanceOf,
+          tokenAddress,
+        ]);
+        const callback = async () => {
+          const contract = new Contract(
+            tokenDetails.address,
+            tokenDetails.abi,
+            getProvider(chainId),
+          );
+          const balance = await contract.balanceOf(tokenAddress);
+          setState({ value: balance, loading: false, error: null });
+        };
+        await asyncCall(hashedArgs, callback, {
+          ttl: 1000 * 30,
+          fallbackToPreviousResult: true,
+        });
+      } catch (e) {
+        setState({ value: '0', loading: false, error: e });
+      }
+    };
+    getBalance();
```

Comment:
Use `useAssetBalance` as example to make use of call cache:

```suggestion
    let sub: Subscription;
    const getBalance = async () => {
        const { address: tokenAddress } = await getTokenContract(
          contractToken,
          chainId,
        );
        const tokenDetails = await getTokenDetails(asset, chainId);

        const hashedArgs = idHash([
          tokenDetails.address,
          EcosystemDataType.balanceOf,
          tokenAddress,
        ]);
        
        sub = observeCall(hashedArgs).subscribe(e => setState(e.result));
        
        const callback = () => {
          return new Contract(
            tokenDetails.address,
            tokenDetails.abi,
            getProvider(chainId),
          ).balanceOf(tokenAddress);
        };
        startCall(hashedArgs, callback, {
          ttl: 1000 * 30,
          fallbackToPreviousResult: true,
        });
    };
    getBalance().catch(e => setState({ value: '0', loading: false, error: e }));
```